woocommerce/woocommerce
# Updated the Jetpack Autoloader to v2.7.1

### All Submissions:


### Changes proposed in this Pull Request:

Over the last five months I have made a number of contributions to the [automattic/jetpack-autoloader](https://github.com/Automattic/jetpack-autoloader) package to resolve bugs, improve performance, and improve developer quality-of-life. This PR bumps that dependency from `v2.2.0` to `v2.7.1` in order to realize the benefit of those improvements in WooCommerce Core. (Technically it bumps it from `v2.6.0` but this is due to a Renovate PR being merged on accident and has not shipped in a WooCommerce release!)

#### Relevant Improvements

- Adds support for Composer v2.0.7 and above.
- Fixes cases where `active_plugins` being filtered breaks the autoloader entirely. (https://github.com/woocommerce/woocommerce/issues/27608)
- Adds a plugin list cache to improve autoloader performance.

I'd like to note that this version also includes unoptimized loading for PSR-4 during development but we can't yet utilize this. I've set the `config.optimize-autoloader` key in `composer.json` for now to ensure we're always optimizing the build as a way to compensate for the fact that our build systems aren't yet set up to install using the `-o` option. Once we've adjusted our build steps to optimize the autoloader we can remove this key and use the unoptimized builds during development.

### How to test the changes in this Pull Request:

1. Make sure that the WooCommerce plugin still works correctly
2. Install "WooCommerce Blocks" and make sure that it works correctly as well. I should suggest checking the SSR and making sure that the version being loaded is correct.

### Other information:




### Changelog entry

> Dev - Updated the Jetpack autoloader to v2.7.1.
